src/evergreen/api.py

```diff
@@ -164,10 +164,23 @@ def _call_api(
         :return: response from api server.
         """
         start_time = time()
+        LOGGER.debug(
+            "Request to be sent",
+            url=url,
+            params=params,
+            timeout=self._timeout,
+            data=data,
+            method=method,
+        )
         response = self.session.request(
             url=url, params=params, timeout=self._timeout, data=data, method=method
         )
 
+        LOGGER.debug(
+            "Response received",
+            response_status_code=response.status_code,
+            response_text=response.text,
+        )
```

Comment:
I think this is fine for debugging purposes, but for the most part we run our services with a log level of info and above. So, these lines often won't say anything, right?

I think it might be helpful in the error case if we logged out something with a higher log level. That way actual errors will by default be bubbled up and won't require us to modify all of our services to log at debug.

I think this can be accomplished by modifying this function some
https://github.com/evergreen-ci/evergreen.py/blob/fad46327f3a9e8dbd7e9e214fa81a37478d73d8a/src/evergreen/api.py#L192

What do people think?

Reply:
If anyone is worried about this being noisy, we also could hide the above feature behind a parameter that's passed in when the client is created. So, people would have to opt into this behavior at client creation time.

https://github.com/evergreen-ci/evergreen.py/blob/fad46327f3a9e8dbd7e9e214fa81a37478d73d8a/src/evergreen/api.py#L73

Reply:
> I think this is fine for debugging purposes, but for the most part we run our services with a log level of info and above. So, these lines often won't say anything, right?
> 
> I think it might be helpful in the error case if we logged out something with a higher log level. That way actual errors will by default be bubbled up and won't require us to modify all of our services to log at debug as that may be pretty normal for everyday usage.
> 
> I think this can be accomplished by modifying this function some
> 
> https://github.com/evergreen-ci/evergreen.py/blob/fad46327f3a9e8dbd7e9e214fa81a37478d73d8a/src/evergreen/api.py#L192
> 
> What do people think?

SG. I changed these to info level and guarded with a debug flag, so the only change needed for these to show up in our application logs would be setting the flag to `True` in constructor.